### PR TITLE
fix: bug with shuffle not getting state before toggle

### PR DESCRIPTION
**Pull request by sjclayton**
Caught a small bug, and figure I'd zap it straight away... ⚡

### CODE REVIEW

File: lua/music-controls/init.lua [hunk 1]
```diff
@@ -243,7 +243,16 @@ M.shuffle = function(player)
...
 
-    exec_command('playerctl -p  '.. player[1] .. ' shuffle toggle')
+    local current_status = exec_command('playerctl -p '.. player[1] .. ' shuffle')
+    sleep(0.25)
+
+    if current_status == 'Off' then
+      exec_command('playerctl -p  '.. player[1] .. ' shuffle toggle on')
+    else
+      exec_command('playerctl -p  '.. player[1] .. ' shuffle toggle off')
+    end
```
Comment (AntonVanAssche) [inline lua/music-controls/init.lua]: Please make sure to use 4 spaces as indentation.

```suggestion
    if current_status == 'Off' then
        exec_command('playerctl -p  '.. player[1] .. ' shuffle toggle on')
    else
        exec_command('playerctl -p  '.. player[1] .. ' shuffle toggle off')
    end
```

**Review by AntonVanAssche — CHANGES_REQUESTED** [review]

File: lua/music-controls/init.lua (hunk 1 above)
Comment (sjclayton) [reply lua/music-controls/init.lua]: I'm going to add a `.editorconfig` file so my auto format on save is proper for this project... I don't like having to pay attention to whether or not I've disabled it before I save.

Also, sometimes I'm not convinced tree-sitter's indent feature works correctly all the time.

**Comment by sjclayton** [comment]
All good, now that can't happen anymore... for me or anyone else who contributes in the future! _lol... shakes head_ 😆 

Just in case you didn't know... now Neovim (works with other editors as well) will automatically use the `.editorconfig` file to override the settings of a contributors editor if they per say prefer 2 space indentation and it will automatically ensure 4 space indentation when they are working within this project.

The `.stylua.toml` does more or less the same thing but for Stylua which is a formatter for Lua, but will only be applied if Stylua is installed and properly configured, ie. it needs something to call it, because it doesn't run automatically.

I looked at your Neovim config, and it appears you have Stylua installed through Mason but you have nothing that is actually calling it to do the formatting, so essentially it's doing nothing... you'd need something like either [none-ls.nvim](https://github.com/nvimtools/none-ls.nvim) or [conform.nvim](https://github.com/stevearc/conform.nvim) (which is excellent btw) to enable Stylua to run on save, or when called from a keymap for example.

**Review by AntonVanAssche — APPROVED** [review]
Looks great! Thanks for this fix.

**Comment by AntonVanAssche** [comment]
> I looked at your Neovim config, and it appears you have Stylua installed through Mason but you have nothing that is actually calling it to do the formatting, so essentially it's doing nothing... you'd need something like either [none-ls.nvim](https://github.com/nvimtools/none-ls.nvim) or [conform.nvim](https://github.com/stevearc/conform.nvim) (which is excellent btw) to enable Stylua to run on save, or when called from a keymap for example.

Thanks for letting me know, I didn't know this. (I don't write Lua that often, besides config stuff 😉)

**Comment by sjclayton** [comment]
> Thanks for letting me know, I didn't know this. (I don't write Lua that often, besides config stuff 😉)

@AntonVanAssche 

Even if that's all you primarily write in Lua, you can copy that `.stylua.toml` into your config directory and then it'll make sure all your configs are formatted and kept in a clean consistent state... I do it with mine, helps if you occasionally format a function slightly wrong or miss commas in plugin specs etc.

What I said about Stylua only working if it has something to call it on save or a keymap (such as none-ls or conform), that same thing applies to any formatting tools... none of them will work unless you have something to actually run them on an action. Think of something like conform like nvim-lint but for formatting. Also conform's configuration is very similar to how nvim-lint configures linters. 😄

**Pull request merged**